lib/github_service/commands/run_test.rb

```diff
@@ -0,0 +1,302 @@
+        push_commit_to_remote
+        create_cross_repo_test_pull_request
+      end
+
+      private
+
+      def _execute(issuer:, value:)
+        @issuer = issuer
+        parse_value(value)
+        return unless valid?
+
+        run_tests
+      end
+
+      def parse_value(value)
+        @value = value
+
+        @test_repos, @repos = value.split(/\s+including\s+/)
+                                   .map { |repo_list| repo_list.split(",") }
+
+        # Add the identifier for the PR for this comment to @repos here
+        @repos ||= []
+        @repos  << "#{issue.repo_name}##{issue.number}"
+
+        @test_repos.map! { |repo_name| normalize_repo_name(repo_name) }
+        @repos.map!      { |repo_name| normalize_repo_name(repo_name) }
+      end
+
+      def normalize_repo_name(repo)
+        repo.include?("/") ? repo : "#{issue.organization_name}/#{repo}"
+      end
+
+      def valid?
+        unless issue.pull_request?
+          issue.add_comment("@#{issuer} 'run-test(s)' command is only valid on pull requests, ignoring...")
+          return false
+        end
+
+        unless repos_valid?(@test_repos + @repos)
+          issue.add_comment("@#{issuer} '#{value}' is an invalid repo, ignoring...")
+
+          return false
+        end
+
+        true
+      end
+
+      REPO_ONLY_REGEXP = /(?:[^\/]+\/)?(?<REPO_NAME_ONLY>[^@#]+)/.freeze
+      def repos_valid?(repos)
+        repos.all? do |repo|
+          name_only = repo.match(REPO_ONLY_REGEXP)[:REPO_NAME_ONLY]
+          GithubService.org_repos(issue.organization_name).include?(name_only)
+        end
+      end
+
+      ##### run_tests steps #####
+
+      # Clone repo (if needed) and initialize @rugged_repo
+      def clone_repo_if_necessary
+        repo_path = self.class.test_repo_clone_dir
+        if Dir.exist?(self.class.test_repo_clone_dir)
+          @rugged_repo = Rugged::Repository.new(repo_path)
+        else
+          url = self.class.test_repo_url
+          @rugged_repo = Rugged::Repository.clone_at(url, repo_path)
+        end
+        git_fetch
+      end
+
+      def create_cross_repo_test_branch
+        rugged_repo.create_branch(branch_name, "origin/master")
+        rugged_repo.checkout(branch_name)
+      end
+
+      def update_travis_yaml_content
+        Dir.chdir(@rugged_repo.workdir) do
+          content = YAML.load_file(".travis.yml")
+
+          content["env"] = {} unless content["env"]
+          content["env"]["global"] = ["REPOS=#{repos.join(',')}"]
+          content["env"]["matrix"] = test_repos.map { |repo| "TEST_REPO=#{repo}" }
+
+          File.write('.travis.yml', content.to_yaml)
+        end
+      end
+
+      def commit_travis_yaml_changes
+        index = rugged_repo.index
+        index.add('.travis.yml')
+        index.write
+
+        # rubocop:disable: Rails/TimeZone
+        bot       = self.class.bot_name
+        author    = { :name => issuer, :email => "no-name@example.com", :time => Time.now }
```

Comment:
So just a note:  `Rugged` requires the email here, so I was just filling in a dummy example here.

In reality, it would be cool if the email registered from the user was used for this, since it would allow documenting in the commit both the bot creating it, and the user that triggered it.

That said, it is just a "cute" feature, and not really a hard requirement.

Reply:
The bot actually has an email, so we can use that if we make it a setting

Reply:
Oh I see you have the bot email below...yeah, let's make the bot email a setting (it's actually a gmail address).

For the author bit here, yeah, it would be kind of cool if we could get the comment requester's email, but it might be weird that one can get "commit credit" for something they didn't actually author.  Not sure how I feel about that :)

Reply:
Well, it is similar to how they do email based merges, where someone sends an email of a patch, and the maintainer applies and merges it themselves.  In that case, the "author" is the person who wrote the patch (in this case, the person triggering the command), and the "committer" is the bot, who is actually doing the "merge".